### ISSUE TITLE
Make logits_to_keep a runtime input (cheap prefill plus speculative decode from one export)

### ISSUE THREAD

**Issue by JulienBalianSonos** [sonos/torch-to-nnef#216]
## Summary

Make `logits_to_keep` (aka `num_logits_to_keep`) a runtime input instead of a baked constant. Today we pass a Python int at wrap time, so the last-token slice is fixed in the graph. One export serves one regime only.

## Why

Prefill and speculative decode want different slicing.

Prefill wants `logits_to_keep = 1`. We feed `S` prompt tokens but only need the last row. Projecting the LM head over all `S` rows is wasted work. On a large vocab (Qwen3 is 152k) and a long prompt this is a big matmul, on the order of `S x hidden x vocab`.

Speculative decode wants `logits_to_keep = k + 1` for `k` draft tokens. It needs per position logits, but only over a few rows, so it is cheap.

A static `--num-logits-to-keep 0` export gives speculative what it needs but taxes every prefill. A runtime value serves both from a single file.

It also drops the tract side workaround (`expose_all_logits` plus a raw non decluttered load), since the model already returns the right number of rows per call. See sonos/tract#2370 and sonos/tract#2381.

## Use case

Export once:

```python
exporter = LLMExporter(model, tokenizer, num_logits_to_keep="dynamic")
```

Then two call shapes, same model:

```
# prefill: long prompt, 1 row out
run(input_ids=[1, S], logits_to_keep=1)        -> logits [1, 1, vocab]

# speculative verify: tail + k drafts, k+1 rows out
run(input_ids=[1, tail+k], logits_to_keep=k+1) -> logits [1, k+1, vocab]
```

Same NNEF, no re-export, no prefill tax.

## Open points

Thread a symbolic dim through the `lm_head` slice (`[:, -K:, :]` with `K` a runtime input) instead of a fixed int. Confirm tract handles a dynamic slice length on that axis end to end. Keep the current static behavior as default and make dynamic opt-in.
